### DISCUSSION TITLE
Navigation simulation with long press

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#399] [Q&A] [answered]
"Simulate navigation" is currently accessible only via some clicks (Menu > Settings > Navigation > Extended > Navigation). My suggestion is to provide a "Simulate navigation" (or "Simulate route") option by long-pressing the routing button (similar to long-press routing button > Stop navigation).

<img src="https://github.com/devemux86/cruiser/assets/145460147/64744c5f-3584-4ba2-94c2-0f33db17bc5b" width="33%"/>

_Originally posted by @bios4 in https://github.com/devemux86/cruiser/discussions/397#discussion-5966191._

**Comment by DolomitesTom** [comment]
Why not? This position is not occupied at the moment.

**Reply by WolfgangZ42** [reply]
That's not true - navigation starts with a long tap too ...

**Reply by devemux86** [reply]
> navigation starts with a long tap too

This is just a side effect because there is no long press enabled in the navigation button.

**Comment by Ciutnik** [comment]
Isn't it a pity to "waste" such an easily accessible action on a rarely used function? Of course, I do not deny that simulation is necessary, but does it necessarily have to be easily accessible? 
~~Maybe use the long press for something useful on the route – e.g. a detour or force a recalculation or something like that.~~

Edit: Ah yes, sorry, what I'm writing about is available from the navigation panel, and the @bios4 suggests using routing button before starting navigation.

**Reply by devemux86** [reply]
It is about the navigation button at bottom-left with icon: ![Navigation](https://github.com/devemux86/cruiser/assets/3484020/a32c6015-9308-4684-96d5-22430dfef33f)

@Ciutnik that's correct, it is before starting navigation when long press is available.

**Reply by bios4** [reply]
> After navigation starts, the long press is empty again.

Uhhm - no!??!
A long-press AFTER start of navigation brings up the "Stop navigation" dialog:

<img src="https://github.com/devemux86/cruiser/assets/145460147/bf4c1177-ffab-4fd4-b3fb-736d099fff04" width="33%"/>

That's why I thought that offering a "Simulate navigation" (or "Simulate route") screen in the very same way BEFORE navigation is started would be a nice addition. 😉

**Reply by devemux86** [reply]
@bios4 thanks, that's right! :slightly_smiling_face: 
(I edited my post)

**Comment by devemux86** [comment]
Navigation simulation with long press is coming in next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/08cf8546-4e7f-4312-80b1-48b1930af97a" width="50%"/>

**Reply by bios4** [reply]
This is awesome!

May I suggest to name it "Simulate navigation" (or "Simulate route") and "Cancel simulation"?

And for the German translation "Navigation simulieren" (or "Route simulieren") and "Simulation abbrechen" (to cancel) would be nice...

👍

**Reply by devemux86** [reply]
Do we need a question when we stop the simulation?
Currently is can be stopped by just tapping again the navigation button.

**Reply by bios4** [reply]
I would implement it in the very same way as start/stop navigation - for consistency, as well as for ease of use (also especially for new users)...

**Reply by devemux86** [reply]
Note that the long press to stop navigation is needed because the tap is used to pause / resume navigation.

Simulation does not have a pause function, so the tap is free when simulation is active.

**Comment by Ciutnik** [comment]
If stimulation will be activated by a long press, we can probably abandon menu `Advanced | Navigation`. 
So it can move this whole menu from the Settings menu to the long press menu. We would have all three types here that we have now: Normal, Simulation, Static.

**Reply by devemux86** [reply]
> we can probably abandon menu Advanced | Navigation.

Correctly, but without static, it seems a bit complicated for most users especially in main buttons. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> Correctly, but without static

Well, that's a bit of a pity. This was very cool, maybe an underrated feature. But maybe you're right that it's not worth maintaining complicated functions that only a few people use 🙂

**Reply by devemux86** [reply]
Also important is that we move something in the foreground where all users can easily find it (instead of advanced settings).
3 options for navigation is a bit too much for most users, they will get confused. :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
Navigation simulation with long press on navigation button is available in [Cruiser 3.0.37](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by ramondevic** [reply]
I can't find how to activate the simulation, it has been removed from the settings menu but I don't see how it can be used.

**Reply by devemux86** [reply]
@ramondevic 

Long press the navigation button and a menu opens to select navigation or simulation.

**Comment by ramondevic** [comment]
I can't find it, I don't know which the button is.
Navigation button is direction button?

**Reply by devemux86** [reply]
@ramondevic it is this button:

![Navigation](https://github.com/devemux86/cruiser/assets/3484020/0d89d6ec-13ee-4e6f-888d-1b788a828b8c)

**Comment by ramondevic** [comment]
I had tried it anyway... except pressing the button right after loading the route.

When the navigation has started, it no longer goes out: neither with the navigable route, nor by stopping the navigation, nor by recalculating the route.
You need to delete the route or import it again for it to work.

Thanks !

**Reply by devemux86** [reply]
@ramondevic do you mean how to stop navigation?

- **Long press** the navigation button to **stop** navigation.
- **Tap** the navigation button to **pause** navigation.

**Reply by ramondevic** [reply]
To stop navigation, I used to just click, now I have learned that I had to make a long click.

**Reply by devemux86** [reply]
@ramondevic do you mean simulation?

- Navigation always had to be stopped with long press.
(tap is to pause navigation)

- Simulation now needs long press to stop it (similar to navigation).

**Reply by ramondevic** [reply]
Of course, I didn't know.
Thanks.

**Reply by 0709wiwiwi** [reply]
So maybe an optimisation is possible after all.
Highlight buttons and active top panels with long-press function.
Active panels: Small discreetly coloured triangles in the corners.
_A discrete coloured outer marker might also make sense. 
But  I suspect not everyone wants such a presentation._
Buttons: A small coloured outer ring.
